crates/viewer/re_renderer/src/video/decoder/web.rs

```diff
@@ -300,20 +295,37 @@ impl VideoDecoder {
     }
 
     fn try_present_frame(&mut self, presentation_timestamp: Time) -> FrameDecodingResult {
+        re_tracing::profile_function!();
+
+        let mut decoder_output = self.decoder_output.lock();
+
         let timescale = self.data.timescale;
 
-        let mut frames = self.frames.lock();
+        let frames = &mut decoder_output.frames;
 
         let Some(frame_idx) = latest_at_idx(
             &frames,
             |frame| frame.composition_timestamp,
             &presentation_timestamp,
         ) else {
-            // no buffered frames - texture will be blank
+            // No buffered frames - texture will be blank.
+
+            // Might this be due to an error?
+            if let Some(last_decoding_error) = decoder_output.last_decoding_error.clone() {
```

Comment:
question: Doing it this way makes sense to me, but I wonder if we should unconditionally check for errors, instead of only in case we don't have any buffered frames? Would that make a difference in any case?

Reply:
Given that we throw away everything when reseting because of error it really shouldn't make a difference - if we fully recoverd by the time we enter here (impossible since we haven't given control back to the browser) then we wouldn't care about the error no matter how much time passed.
It's more that my internal logic went "only look for errors if we're out of frames" 🤔